### PR TITLE
Update Service GC Safe Point after Flushing

### PR DESCRIPTION
**Pull request by YuJuncen**
This PR:
- Adapted the `update_service_gc_safepoint` for `PdClient`. And implemented this for `RpcClient`. (Maybe we should implement it for more clients for make tests passed.)
- Update a service GC safe point named `br-stream-{task_name}-{store_id}`, and after each time we do flushing, set the current checkpoint TS for it, with `TTL = 10mins`(Maybe longer?).
- Trying to advance the checkpoint TS even there is no new writes. That has been done by :
	1. Get TS from the PD.
	2. Try to advance the Resolved TS by this TS for each region.
	3. Choose `max(min_resolved_ts_of_files.unwrap_or(0), regions_resolved_ts.min())` as the new resolved ts.
- Fix a bug that caused regions with `Follower` role mistakenly registered to be observed.

### CODE REVIEW

**Review by joccau — APPROVED** [review]
LGTM ~

File: components/br-stream/src/observer.rs [hunk 1]
```diff
@@ -229,6 +233,16 @@ mod tests {
 
     use super::BackupStreamObserver;
 
+    macro_rules! assert_let {
```
Comment (kennytm) [inline components/br-stream/src/observer.rs]: wait i've seen this in another PR 🤔

Comment (YuJuncen) [reply components/br-stream/src/observer.rs]: Yes, that PR is actually extract from this PR.

**Review by kennytm — APPROVED** [review]
please also fix https://github.com/3pointer/tikv/pull/26#pullrequestreview-898525303

**Pull request merged**